Exception handling, async fixes, tests

alertedsnake opened the pull request:
Added some exception logging for some cases where we are getting them in real use but it's not clear why.

Added `await` and `async` keywords in cases where they should have been, but weren't.

A number of tests were broken - some were testing functions which didn't exist anymore, and none of the async stuff was tested properly. This is now all fixed.

Fixed up the `test` Make target so it actually installs things properly and runs tests.

Also bumped versions of dependencies and Alpine to modern times.

Commit:
Exception handling, async fixes, tests

Added some exception logging for some cases where we are getting them in
real use but it's not clear why.

Added `await` and `async` keywords in cases where they should have been,
but weren't.

A number of tests were broken - some were testing functions which didn't
exist anymore, and none of the async stuff was tested properly. This is
now all fixed.

Fixed up the `test` Make target so it actually installs things properly
and runs tests.

Also bumped versions of dependencies and Alpine to modern times.

mgundel commented on rssalertbot/feed.py:
```diff
@@ -144,8 +145,6 @@ async def _handle_fetch_failure(self, title, description):
             description (str): alert description
         """
 
-        self.log.error(f"{title}: {description}")
```
I would definitely expect this method to still be logging errors, as the method description says.

alertedsnake replied:
Ah, I can make that more clear.

mgundel commented on rssalertbot/feed.py:
```diff
@@ -127,11 +127,12 @@ async def _fetch(self, session, timeout=10):
                 etype = '.'.join((type(e).__module__, type(e).__name__))
-                self._handle_fetch_failure('Exception', f"{etype} fetching feed: {e}")
+                await self._handle_fetch_failure('Exception', f"{etype} fetching feed: {e}")
+                self.log.exception("Error fetching feed")
```
I think it's fine to log this as `error` via `_handle_fetch_failure` in order to retain the alerting functionality.

alertedsnake replied:
I specifically wanted the exception trace itself to help diagnose this, and wanted to do it outside of another async call. Though it now occurs to me I should swap the order of lines 134 and 135 to log first.

mgundel replied:
In that case, the two other calls to `_handle_fetch_failure` should be updated to log independently of the method call.

Commit:
Re-order exception handling, fix docs

Also caught another silly issue where `pendulum.now()` was called twice
for no reason.

Commit:
Some handling for other errors
